### PR TITLE
Backport: Add parameter server [fix CI] (#88)

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,16 +25,16 @@ jobs:
       image: ${{ matrix.docker_image }}
     steps:
     - uses: actions/checkout@v2
-    - uses: ros-tooling/setup-ros@0.1.2
+    - uses: ros-tooling/setup-ros@0.1.4
```
Comment (JanStaschulat) [inline .github/workflows/ci.yml:28]:
```suggestion
    - uses: ros-tooling/setup-ros@0.2.0
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,16 +25,16 @@ jobs:
...
       with:
         required-ros-distributions: ${{ matrix.ros_distribution }}
     - name : Download and install rclc-dependencies
       run: |
         apt-get install ros-${{ matrix.ros_distribution }}-osrf-testing-tools-cpp
         apt-get install ros-${{ matrix.ros_distribution }}-test-msgs
-    - uses : ros-tooling/action-ros-ci@0.1.0
+    - uses : ros-tooling/action-ros-ci@0.2.0
```
Comment (JanStaschulat) [inline .github/workflows/ci.yml:35]:
```suggestion
    - uses : ros-tooling/action-ros-ci@0.2.1
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JanStaschulat) [inline .github/workflows/ci.yml:28]: See also suggestion of Christophe https://github.com/ros2/rclc/pull/91

**Review by JanStaschulat — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -25,16 +25,16 @@ jobs:
       image: ${{ matrix.docker_image }}
     steps:
     - uses: actions/checkout@v2
-    - uses: ros-tooling/setup-ros@0.1.2
+    - uses: ros-tooling/setup-ros@0.2.0
```
Comment (JanStaschulat) [inline .github/workflows/ci.yml:28]: @pablogs9 maybe you also have to change the two `branches` values to `galactic` as in this pull request: https://github.com/ros2/rclc/pull/91

**Review by JanStaschulat — APPROVED** [review]

**Comment by JanStaschulat** [comment]
@pablogs9 Hmm, the bloom release does not know the branch `galactic` when checking out the code: 
```
22:37:24  > git rev-parse origin/galactic^{commit} # timeout=10
22:37:24 FATAL: Command "git rev-parse origin/galactic^{commit}" returned status code 128:
22:37:24 stdout: origin/galactic^{commit}
22:37:24 
22:37:24 stderr: fatal: ambiguous argument 'origin/galactic^{commit}': unknown revision or path not in the working
```
I asked why this is the case to the guys who manage the rosdistro repository: https://github.com/ros/rosdistro/pull/29683

File: .github/workflows/ci.yml (hunk 3 above)
Comment (pablogs9) [reply .github/workflows/ci.yml:28]: They are already changed I guess: https://github.com/ros2/rclc/blob/6b755f44a4c6c8a83104c142b3fc0eb9651b65e3/.github/workflows/ci.yml#L5

**Comment by JanStaschulat** [comment]
Created new bloom release (request) https://github.com/ros/rosdistro/pull/29702
Let's see if that resolves the errors that 'galactic' branch is not found.

**Comment by pablogs9** [comment]
Cool thanks Jan

**Review by JanStaschulat — APPROVED** [review]

**Comment by pablogs9** [comment]
Should I merge this once CI pass or do you prefer to wait until `Gpr__rclc__ubuntu_focal_amd64` is passing?

**Comment by JanStaschulat** [comment]
> Should I merge this once CI pass or do you prefer to wait until `Gpr__rclc__ubuntu_focal_amd64` is passing?

Let's wait until tomorrow  for the  `Gpr__rclc__ubuntu_focal_amd64` to pass. I created a new bloom release, in which i changed the upstream version to `galactic` and posted the questions in the PR at `rosdistro`. I would like to give them a possibility to reply until this pull request is still open.

**Comment by pablogs9** [comment]
Ok

**Comment by JanStaschulat** [comment]
We need to updatede the GPG key in the build job somehow:
https://discourse.ros.org/t/ros-gpg-key-expiration-incident/20669

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-gpg-key-expiration-incident/20669/20

**Comment by JanStaschulat** [comment]
Wrote a question at ros-discours, how to update ci-jobs : https://discourse.ros.org/t/ros-gpg-key-expiration-incident/20669/20

**Comment by pablogs9** [comment]
@JanStaschulat the `ros-tooling` issue is solved by increasing the version. Was the `Gpr__rclc__ubuntu_focal_amd64` error related to the bloom release?

**Comment by JanStaschulat** [comment]
> 
> 
> @JanStaschulat the `ros-tooling` issue is solved by increasing the version. Was the `Gpr__rclc__ubuntu_focal_amd64` error related to the bloom release?

Yes, there is an error with the configuration of the bloom release. My question in the PR to rosdistro was not answered. So i created a new issue https://github.com/ros/rosdistro/issues/29738
@christophebedard Do you have any idea, how to resolve this?

**Comment by christophebedard** [comment]
> Yes, there is an error with the configuration of the bloom release. My question in the PR to rosdistro was not answered. So i created a new issue [ros/rosdistro#29738](https://github.com/ros/rosdistro/issues/29738)
> @christophebedard Do you have any idea, how to resolve this?

sorry, I don't know. I looked at the rosdistro config for `rclc` and the release repo config/tracks and it all seems okay to me.

**Comment by JanStaschulat** [comment]
@Karsten1987 Do you have any idea, how to resolve the build-farm error of the bloom release for Galactic 
https://github.com/ros/rosdistro/issues/29738?

**Comment by pablogs9** [comment]
@JanStaschulat we at eProsima wanted to have micro-ROS Galactic ready tomorrow, in our opinion this can be merged since the CI problem is not related to params. Are you ok with merging and opening another PR to solve the bloom issue?

**Comment by JanStaschulat** [comment]
> 
> 
> @JanStaschulat we at eProsima wanted to have micro-ROS Galactic ready tomorrow, in our opinion this can be merged since the CI problem is not related to params. Are you ok with merging and opening another PR to solve the bloom issue?

@pablogs9 Yes, you merge this PR, but please opne an issue or another PR to solve the bloom release problem. I also think this is not an issue with rclc_parameter  pull request, but due to a configuration error at the build farm. Nobody has responded yet, unfortunately. So we keep it out of our critical path.

**Comment by pablogs9** [comment]
Thanks, @JanStaschulat. I'm merging and opening another dummy PR for solving the bloom issue.

**Pull request merged**